### ISSUE TITLE
Android (ExoPlayer) cannot force TCP for RTSP

### ISSUE THREAD

**Issue by machenmusik** [RenderHeads/UnityPlugin-AVProVideo#1619]
For Android playback, ExoPlayer supports forcing TCP for RTSP `setForceUseRtpTcp(true)`

However, there does not seem to be a way to do that via AVPro, 
and ExoPlayer does not correctly fall back to TCP when UDP is unavailable.

This combination causes playback for RTSP/TCP-only sources to fail.

Ideallly, the ability to force TCP would be exposed as an Android-only setting, similar to ExoPlayer buffer sizes.

**Comment by Ste-RH** [comment]
We can look into exposing this in some way.  Do you have a stream URL you can share that we can test against?  If you wish to keep it private, you can email it over to unitysupport@renderheads.com with the subject 'Issue 1619'.

**Comment by machenmusik** [comment]
Easiest thing is to set up a RTSP server that only accepts TCP and rejects UDP (and multicast).
For example, see https://github.com/bluenviron/mediamtx#transport-protocols for their syntax (you don't need to use encryption)

**Comment by machenmusik** [comment]
The changes are pretty straightforward to do this, in the exoplayer java bridge, let me know if you'd like to discuss in detail.

**Comment by Ste-RH** [comment]
We have added this into the next release of AVPro Video v2.8.5.  This should hopefully be released today.  To action it, there is an option in the Android platform properties of the MediaPlayer component.

**Comment by Ste-RH** [comment]
v2.8.5 has been released @machenmusik.  Do let us know if the change does what you require.

**Comment by machenmusik** [comment]
Works thank you!

Question, can / does AVPro use ExoPlayer's async mode (no longer experimental) ? https://stackoverflow.com/questions/75947834/android-exoplayer-cannot-see-new-method-of-operation-mode-in-defaultrenderersfac

**Comment by Ste-RH** [comment]
> Works thank you!
> 
> Question, can / does AVPro use ExoPlayer's async mode (no longer experimental) ? https://stackoverflow.com/questions/75947834/android-exoplayer-cannot-see-new-method-of-operation-mode-in-defaultrenderersfac

No.  We had it enabled for a period of time in our local testing environments when it was experimental, but it was too risky to release enabled.  Now that it is officially stable though, we will look to add an option to force it enabled in the next release.
